Pod/Classes/GraphQL/ARGraphQLQueryCache.m

```diff
@@ -158,7 +158,9 @@ - (void)clearAll;
     if (response == nil) {
         // this is a sentinel to indicate the value is being fetched and a place where interested parties can request a
         // future resolved value as a promise
-        NSAssert(self.inFlightRequests[cacheKey] == nil, @"[ARGraphQLQueryCache] [%@] Expected no promise queue to exist yet", cacheKey);
+        if (self.inFlightRequests[cacheKey] == nil) {
+            DLog(@"[ARGraphQLQueryCache] [%@] Expected no promise queue to exist yet", cacheKey);
```

Comment:
this `NSAssert` actually shows up as a red screen in React Native which is the error we were seeing regularly.

Reply:
![](https://files.slack.com/files-pri/T02531TU5-FFR5SJ05A/screen_shot_2019-01-28_at_3.30.32_pm.png)

Reply:
I ran into this assertion failure while trying to do prefetching for bid flow. I didn't have the chance to go back and dig into it. @alloy does this make change make sense?

Reply:
This just works around the assertion being hit, there's definitely something off with the overall codeflow if this is being hit. I'm fine with falling back to the same code that's used in production to ease development, for now, but a ticket should be filed to look into this imo.

Reply:
Gotcha, I've opened a ticket here to track: https://artsyproduct.atlassian.net/browse/GROW-1098

Reply:
Thanks! 👍